streamz/core.py

```diff
@@ -319,18 +339,35 @@ def remove_stream(change):
 
         return output._ipython_display_(**kwargs)
 
-    def _emit(self, x):
+    def _emit(self, x, metadata=None):
+        if metadata is None:
+            metadata = {}
+
+        refs = []
+        if 'refs' in metadata:
+            refs = metadata['refs']
+            if not isinstance(refs, list):
+                refs = [refs]
+
         result = []
-        for downstream in list(self.downstreams):
-            r = downstream.update(x, who=self)
+        downstreams = list(self.downstreams)
```

Comment:
This line is not necessary, just refer to `self.downstreams` in the lines below. `len()` and iteration will work regardless.

Reply:
fixed.